examples/auth/auth_service.exs

```diff
@@ -0,0 +1,95 @@
+# Run this script from the Spell root with
+#
+#     mix spell.example.auth
+
+defmodule Auth do
+  @moduledoc """
+  The `RPC` module implements the example for asynchronous Remote Procedure Call.
+  """
+
+  defmodule Callee do
+    require Logger
+    defstruct [:callee, :procedure, timeout: 1000]
+
+    # Public Functions
+
+    @doc """
+    Start the callee with the passed procedure.
+    """
+    def start_link(topics, options \\ []) do
+      {:ok, spawn_link(fn -> init(topics, options) end)}
+    end
+
+    # Private Functions
+
+    defp init(procedure, options) do
+      {:ok, callee} = Auth.new_peer([Spell.Role.Callee], options)
+      {:ok, _register_id} = Spell.cast_register(callee, procedure)
+
+      %__MODULE__{callee: callee,
+                  procedure: procedure}
+        |> struct(options)
+        |> loop()
+    end
+
+    defp loop(state) do
+      receive do
+        :stop -> :ok
```

Comment:
- Oh yes, good catch, I will change this! 

- For this specific example probably will be very unlikely that it will timeout, but for a generic approach we probably want to set the timeout to `:infinity` to avoid sporadic `:errors`